tests/driver_sx127x/main.c

```diff
@@ -0,0 +1,420 @@
+    sx127x_init_lora_settings((sx127x_t*) nd);
+
+    return 0;
+}
+
+int regs_get_cmd(int argc, char **argv)
+{
+    if (argc <= 1) {
+        puts("usage: get <all | allinline | regnum>");
+        return -1;
+    }
+
+    if (strcmp(argv[1], "all") == 0) {
+        puts("- listing all registers -");
+        uint8_t reg = 0, data = 0;
+
+        /* Listing registers map */
+        puts("Reg   0  1  2  3  4  5  6  7  8  9  A  B  C  D  E  F");
+        for (unsigned i = 0; i <= 7; i++) {
+            printf("0x%02X ", i << 4);
+
+            for (unsigned j = 0; j <= 15; j++, reg++) {
+                data = sx127x_reg_read(&sx127x, reg);
+
+                printf("%02X ", data);
+            }
+
+            puts("");
+        }
+
+        puts("-done-");
+
+        return 0;
+    }
+    if (strcmp(argv[1], "allinline") == 0) {
+        puts("- listing all registers in one line -");
+        /* Listing registers map */
+        for (uint16_t reg = 0; reg < 256; reg++) {
+            printf("%02X ", sx127x_reg_read(&sx127x, (uint8_t) reg));
+        }
+
+        puts("-done-");
+
+        return 0;
+    } else {
+        long int num = 0;
+
+        /* Register number in hex */
+        if (strstr(argv[1], "0x") != NULL) {
+            num = strtol(argv[1], NULL, 16);
+        } else {
+            num = atoi(argv[1]);
+        }
+
+        if (num >= 0 && num <= 255) {
+            printf("[regs] 0x%02X = 0x%02X\n",
+                   (uint8_t) num,
+                   sx127x_reg_read(&sx127x, (uint8_t) num));
+        } else {
+            puts("regs: invalid register number specified");
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+int regs_set_cmd(int argc, char **argv)
+{
+    if (argc <= 2) {
+        puts("usage: set <num> <value>");
+        return -1;
+    }
+
+    long num, val;
+
+    /* Register number in hex */
+    if (strstr(argv[1], "0x") != NULL) {
+        num = strtol(argv[1], NULL, 16);
+    } else {
+        num = atoi(argv[1]);
+    }
+
+    /* Register value in hex */
+    if (strstr(argv[2], "0x") != NULL) {
+        val = strtol(argv[2], NULL, 16);
+    } else {
+        val = atoi(argv[2]);
+    }
+
+    sx127x_reg_write(&sx127x, (uint8_t) num, (uint8_t) val);
+
+    return 0;
+}
+
+int send_cmd(int argc, char **argv)
+{
+    if (argc <= 1) {
+        puts("usage: send <payload>");
+        return -1;
+    }
+
+    printf("tx_test: sending \"%s\" payload (%d bytes)\n",
+           argv[1], strlen(argv[1]) + 1);
+
+    struct iovec vec[1];
+    vec[0].iov_base = argv[1];
+    vec[0].iov_len = strlen(argv[1]) + 1;
+    nd->driver->send(nd, vec, 1);
+
+    /* wait for the chip */
+    xtimer_usleep(10000);
+
+    puts("tx_test: sended");
+
+    return 0;
+}
+
+int recv_cmd(int argc, char **argv)
+{
+    nd->driver->set(nd, NETOPT_LORA_SINGLE_RECEIVE, false, sizeof(uint8_t));
+    sx127x_set_rx(&sx127x);
+
+    return 0;
+}
+
+int config_channel_cmd(int argc, char **argv)
```

Comment:
I think we need more verbose for this one
For example we need to set a random argv[1]  to set a channel